src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_sdfg.py

```diff
@@ -99,8 +103,17 @@ def visit_Program(self, node: itir.Program) -> dace.SDFG:
 
         # visit one statement at a time and put it into separate state
         for i, stmt in enumerate(node.body):
+            # I am not sure if it is a good idea to keep the exit state fixed, i.e. adding new states just before it.
```

Comment:
I see a reason for not having the `exit_state` fixed. I would let `visit` add its different states and here, at return from `visit`, I would implement a sanity check that the state machine has a single exit state.

Reply:
You could also do it a bit differently.
If you assume that you know the previous exit state you could just make some graph traversal, which is super cheap since it only involves the new part, and search for the new exit state, which must be uniquely defined.